src/main/java/me/kavin/piped/server/handlers/auth/AuthPlaylistHandlers.java

```diff
@@ -148,22 +148,45 @@ public static byte[] renamePlaylistResponse(String session, String playlistId, S
             ExceptionHandler.throwErrorResponse(new AuthenticationFailureResponse());
 
         try (Session s = DatabaseSessionFactory.createSession()) {
-            var playlist = DatabaseHelper.getPlaylistFromId(s, playlistId);
+            var playlist = PlaylistHelpers.getUserPlaylist(s, user, playlistId);
+            playlist.setName(newName);
 
-            if (playlist == null)
-                return mapper.writeValueAsBytes(mapper.createObjectNode()
-                        .put("error", "Playlist not found"));
+            var tr = s.beginTransaction();
+            s.merge(playlist);
+            tr.commit();
 
-            if (playlist.getOwner().getId() != user.getId())
-                return mapper.writeValueAsBytes(mapper.createObjectNode()
-                        .put("error", "You do not own this playlist"));
+        } catch (IllegalArgumentException e) {
+            return mapper.writeValueAsBytes(mapper.createObjectNode()
+                    .put("error", e.getMessage()));
```

Comment:
We should ideally throw an `ErrorResponse` class instead of returning here so that we can modify the status code.